libfeed/feed-store-provider.c

```diff
@@ -0,0 +1,1239 @@
+      else if (g_strcmp0 (interface_name, "com.endlessm.DiscoveryFeedWord") == 0)
+        g_ptr_array_add (word_proxies, ka_proxy);
+      else if (g_strcmp0 (interface_name, "com.endlessm.DiscoveryFeedQuote") == 0)
+        g_ptr_array_add (quote_proxies, ka_proxy);
+    }
+
+  zipped_word_quote_proxies = zip_two_ptr_arrays (word_proxies,
+                                                  quote_proxies,
+                                                  (PtrArrayZipElementFactory) word_quote_proxy_bundle_new,
+                                                  (GDestroyNotify) word_quote_proxy_bundle_free);
+
+  for (i = 0; i < zipped_word_quote_proxies->len; ++i)
+    {
+      WordQuoteProxyBundle *bundle = g_ptr_array_index (zipped_word_quote_proxies, i);
+      append_discovery_feed_word_quote_from_proxies (bundle->word_proxy,
+                                                     bundle->quote_proxy,
+                                                     cancellable,
+                                                     individual_task_result_completed,
+                                                     individual_task_result_closure_new (all_tasks_results_closure_allocate_new_result_slot (all_tasks_closure),
+                                                                                         all_tasks_closure));
+    }
+}
+
+typedef struct _UnorderedResultsFromQueriesData
+{
+  EosDiscoveryFeedGjsStyleGDBusCallback callback;
+  gpointer                              user_data;
+} UnorderedResultsFromQueriesData;
+
+static UnorderedResultsFromQueriesData *
+unordered_results_from_queries_data_new (EosDiscoveryFeedGjsStyleGDBusCallback callback,
+                                         gpointer                              user_data)
+{
+  UnorderedResultsFromQueriesData *data = g_new0 (UnorderedResultsFromQueriesData, 1);
+
+  data->callback = callback;
+  data->user_data = user_data;
+
+  return data;
+}
+
+static void
+unordered_results_from_queries_data_free (UnorderedResultsFromQueriesData *data)
+{
+  g_free (data);
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (UnorderedResultsFromQueriesData,
+                               unordered_results_from_queries_data_free);
+
+static void
+received_all_unordered_card_array_results_from_queries (GObject      *source,
+                                                        GAsyncResult *result,
+                                                        gpointer      user_data)
+{
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(GPtrArray) results = g_task_propagate_pointer (G_TASK (result),
+                                                           &local_error);
+  g_autoptr(GPtrArray) result_arrays = g_ptr_array_new_with_free_func ((GDestroyNotify) g_ptr_array_unref);
+  g_autoptr(GPtrArray) flat_mapped_array = NULL;
+  g_autoptr(UnorderedResultsFromQueriesData) data = user_data;
+  guint i = 0;
+
+  /* This basically shouldn't happen, but handle it anyway */
+  if (result == NULL)
```

Comment:
Shouldn't happen in the sense that this is like allPendingPromises in main.js - it returns tuples of (error, results) and the caller can handle each error independently - it never throws itself.